modules/diskq/qdisk.c

```diff
@@ -1440,8 +1440,13 @@ _load_qdisk_file(QDisk *self, const gchar *filename, GQueue *qout, GQueue *qback
 static gboolean
 _init_qdisk_file(QDisk *self)
 {
-  self->file_size = QDISK_RESERVED_SPACE;
-  return _create_header(self);
+  if (!_create_header(self))
+    return FALSE;
+
+  if (!self->file_size)
+    self->file_size = QDISK_RESERVED_SPACE;
+
+  return TRUE;
```

Comment:
I would appreciate some a more elaborate comment on why this fixes file_size after prealloc.

Reply:
Good point, I have added details to the commit:
```
    qdisk: fix setting file_size after prealloc

    Setting file_size must be done after each operation, which can alter the
    file's size. Creating the header on the disk is an operation like that,
    but we cannot blindly set the file_size to the header's size, because
    we might have preallocated the file before that which could have already
    set file_size.

    This commit fixes this issue by checking if file_size is already set or
    not, and only setting it to the header's size, when necessary.
```